### DISCUSSION TITLE
Modbus communication error

### DISCUSSION THREAD

**Discussion by Scot787** [artisan-roaster-scope/artisan#889] [Q&A]
I'm running the latest and greatest MacBook Pro. Your latest software and a Buckeye BC-2 drum roaster.

I am intermittently getting a Modbus comm error. With this I lose my BT and ET readout. It often happens when I turn on and off the gas with the igniter. I have to close the software. When I open it and go to config device, TC4 has been deselected and meter Modbus is selected. Then I have to go through the machine setup again. Most times i'm unable to select the correct comm port. I go back to device setup and sure enough the TCP has been automatically deselected. Any suggestions?

**Comment by poundy** [comment]
This isn't a software problem, you realise? 

The igniter causes high voltage spikes, that are affecting the USB connection to your MBP. 

You could look at upgrading your cable, but I am not convinced that this will help (more on that after this)

I don't understand your comments about TC4 / Modbus selections and TCP being deselected. Are you saying that the "fault" changes those settings?  There should be no difference before and after this issue.... Your machine setup should NOT need to be re-run (and that is probably what is causing the TCP change imho) so stop doing that... If the electrically induced fault then causes your COM port number to change on the MBP, then you should work to resolve that.  

Now, cable discussions.  Often, replacing the cable is seen as a good option, but I have reservations in this case, based on what I know of the Buckeye (Blueking) roasters since I have a Blueking one that's imported into AU....  The cable itself is most likely only a standard USB cable, like a printer cable, and most likely, the USB host device is inside the roaster. If the igniter is causing a glitch, it's likely causing the USB host device inside the roaster to reset, and the cable is still passing through the correct signals (but the MBP sees the reset as a "new" device).  If there's any doubt about the cable then it'd be prudent to replace it, but I unfortunately think this is the kind of issue that's not going to be easy to address. 

Has this ever worked reliably? Have you talked to the supplier about the issue? Do you have access to a different PC that you can try and see if it behaves differently?

**Comment by Scot787** [comment]
I am not an expert on computers but I am running a Macbook pro. See the
attached image. I know I'm supposed to have TC4 selected in the device
setup. But for whatever reason it keeps changing itself back to meter. You
say to stop re-running the setup but the only way to gain access to the
device assignment is to run the setup over. Those pages are grayed out in
the config drop down menu. I have to start over to gain access to that
page. I talked to Buckeye and they said it's your software. I'm assuming
I'm the problem and I don't have something set correctly. I'm a huge
audiophile so I'm no stranger to USB cables and jitter. I know I have a
bluetooth option with my roaster. I guess that's my next option. Is it
possible that the jitter from the USB cable could be resetting the
selections to their default settings?

Scott

On Mon, May 23, 2022 at 6:45 PM poundy ***@***.***> wrote:

> This isn't a software problem, you realise?
>
> The igniter causes high voltage spikes, that are affecting the USB
> connection to your MBP.
>
> You could look at upgrading your cable, but I am not convinced that this
> will help (more on that after this)
>
> I don't understand your comments about TC4 / Modbus selections and TCP
> being deselected. Are you saying that the "fault" changes those settings?
> There should be no difference before and after this issue.... Your machine
> setup should NOT need to be re-run (and that is probably what is causing
> the TCP change imho) so stop doing that... If the electrically induced
> fault then causes your COM port number to change on the MBP, then you
> should work to resolve that.
>
> Now, cable discussions. Often, replacing the cable is seen as a good
> option, but I have reservations in this case, based on what I know of the
> Buckeye (Blueking) roasters since I have a Blueking one that's imported
> into AU.... The cable itself is most likely only a standard USB cable, like
> a printer cable, and most likely, the USB host device is inside the
> roaster. If the igniter is causing a glitch, it's likely causing the USB
> host device inside the roaster to reset, and the cable is still passing
> through the correct signals (but the MBP sees the reset as a "new" device).
> If there's any doubt about the cable then it'd be prudent to replace it,
> but I unfortunately think this is the kind of issue that's not going to be
> easy to address.
>
> Has this ever worked reliably? Have you talked to the supplier about the
> issue? Do you have access to a different PC that you can try and see if it
> behaves differently?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/artisan-roaster-scope/artisan/discussions/889#discussioncomment-2807762>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZJTPXWCWF33IU74SLVR443VLQC6XANCNFSM5WXC275Q>
> .
> You are receiving this because you authored the thread.Message ID:
> <artisan-roaster-scope/artisan/repo-discussions/889/comments/2807762@
> github.com>
>


-- 
Scott

**Reply by poundy** [reply]
Hi Scot,
your image doesn't come through. Please use the github page and not respond through email. 

You keep talking about TC4. Unless there's something significantly different in BC roasters, you do NOT have a TC4 and should NOT be setting that. You are incorrect here - there's just no TC4 involved.

I exclusively use the bluetooth connection that my roaster came with (they are customised for Australian delivery from Blueking, so some specs might be different than they deliver to BC Roasters in the US).  So if you have that option, heck, jump on it and be done with cables.  But I would suggest that this might not be a resolution in this case, if the HV spikes that the igniter is causing is actually resetting the hub.

The "usual" USB cable "problem" we see is actually more a ground loop issue on some roasters and laptop combinations causing problems reading the thermocouples, but in this scenario you're talking to the modbus controller inthe roaster so it's unlikely to be related.
  
So what can you do right now. First, please save a copy of your configuration settings that work (so after setup) and save it / rename it with a .txt extension. Drag and drop it onto the github response field so it'll add it here correctly. Also attach the images you had - hopefully that's the "setup" pages before an issue, and after an issue, where we can see your problem.

**Reply by poundy** [reply]
![image](https://user-images.githubusercontent.com/279966/169936764-2ca804a2-88e5-46ab-a40c-8d98cef8b655.png)
![image](https://user-images.githubusercontent.com/279966/169936826-be55f71b-c39e-4873-942f-f5b3623ea8c0.png)
These are the two relevant setup pages you want to screen shot for us.  These are from my working roaster - the only thing different in yours might be the COMM port that is selected.  The first image comes from the CONFIG menu, DEVICE option, and the second image is Config --> Port.

**Reply by poundy** [reply]
and a final answer on your "USB jitter" question, the answer should be no, there's no way a USB jitter of some kind will initiate a reconfiguration of Artisan settings. And I don't think your settings are wrong, I just think that for some reason MacOS has decided to change the comm port port number somehow, so it is no longer the correct one.

I suspect one of the things you are doing in trying to resolve this is you have an active connection (Artisan set to ON so it's attempting to communicate to the roaster) and that would be why the menu items for Device and Port show as greyed out. hit STOP on the roast and menus should be available.

**Comment by Scot787** [comment]
<img width="1440" alt="Screen Shot 2022-05-23 at 23 04 43" src="https://user-images.githubusercontent.com/106117086/169940957-c3f931f7-2507-4113-80f8-069c6088aecc.png">
<img width="1440" alt="Screen Shot 2022-05-23 at 23 04 53" src="https://user-images.githubusercontent.com/106117086/169940965-147dbac4-d6a3-426e-a5cb-23d59f6123e9.png">

**Reply by poundy** [reply]
in my mind, this stood out as "wrong".  The baud rate is not correct, the device path is not what the settings shows, and 
![image](https://user-images.githubusercontent.com/279966/169996459-0dad6bd6-7e3e-44b5-b50d-18e63749e1a9.png)
Can you explain what state the machine and configuration was in when these pictures were taken? 
I'd suggest you ignore igniter on the roaster for a moment and just on getting the software set up correctly. Configure Artisan to talk to the roaster. Save your settings at that point (and attach a copy here !). Then, hit start in Artisan and prove that these settings allow it to track the roaster temps with the gas off and no ignition. Then, at least we know what the correct settings are and that they work. Then you can start a roast, ignite, and all heck breaks loose :) But at least at that point, the comm port will likely reset and it'll have the problem, you can hit STOP and check the settings still look as they were configured only minutes before, and that should again prove that the problem is not in the software
